backend/onyx/connectors/sharepoint/connector.py

```diff
@@ -46,12 +49,75 @@ class SiteDescriptor(BaseModel):
     folder_path: str | None
 
 
+def _sleep_and_retry(query_obj: Any, method_name: str, max_retries: int = 3) -> Any:
```

Comment:
This sleep + retry logic is nice. I don't believe we have a common utility function for this, so maybe in a future PR, we should pull this function out into a more generic file so that we can also reuse it elsewhere.